src/modules/Update.rb

```diff
@@ -801,16 +801,25 @@ def clean_backup
         :force => true, :secure => true)
     end
 
-    # restores backup
+    # Restores valid backups
+    #
+    # It will look for "restore-*.sh" files inside the BACKUP_DIR, executing only those created
+    # after last boot (bsc#1097297)
     def restore_backup
       log.info "Restoring backup"
       mounted_root = Installation.destdir
-      script_glob = File.join(mounted_root, BACKUP_DIR,"restore-*.sh")
+      script_glob = File.join(mounted_root, BACKUP_DIR, "restore-*.sh")
       # sort the scripts to execute them in the expected order
       ::Dir.glob(script_glob).sort.each do |path|
-        cmd = "sh #{path} #{File.join("/", mounted_root)}"
-        res = SCR.Execute(path(".target.bash_output"), cmd)
-        log.info "Restoring with script #{cmd} result: #{res}"
+        backup_time = File.stat(path).ctime.to_i
+
+        if boot_time < backup_time
```

Comment:
I'm not sure if this works on platforms without RTC (e.g. Raspberry Pi). If without RTC the boot time is always 0 then the backup would be always restored. We should check that.